src/Datasource/QueryInterface.php

```diff
@@ -258,14 +257,22 @@ public function page(int $num, ?int $limit = null);
     public function toArray(): array;
 
     /**
-     * Returns the default repository object that will be used by this query,
-     * that is, the repository that will appear in the from clause.
+     * Set the default Table object that will be used by this query
+     * and form the `FROM` clause.
      *
      * @param \Cake\Datasource\RepositoryInterface $repository The default repository object to use
      * @return $this
      */
     public function repository(RepositoryInterface $repository);
 
+    /**
+     * Returns the default repository object that will be used by this query,
+     * that is, the repository that will appear in the from clause.
+     *
+     * @return \Cake\Datasource\RepositoryInterface|mixed $repository The default repository object to use
+     */
+    public function getRepository();
```

Comment:
Since the setter uses typehints, it might be better to also use typehint here: `: RepositoryInterface`

Reply:
Won't that cause phpstan to whine when callers use methods outside of the repository interface?

Reply:
But how is the setter working then?

Reply:
I don't think we call the setter often.